### ISSUE TITLE
'no-unused-vars' shows varsIgnorePattern when argsIgnorePattern is not specified

### ISSUE THREAD

**Issue by akouryy** [eslint/eslint#9774]
'no-unused-vars' shows varsIgnorePattern when argsIgnorePattern is not specified.

**Tell us about your environment**

* **ESLint Version:**
4.13.1 and 4.14.0
* **Node Version:**
8.1.3
* **npm Version:**
5.6.0

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```yml

env:
  browser: true
  es6: true
extends: 'eslint:recommended'
parserOptions:
  sourceType: module
rules:
  eqeqeq: error
  indent:
    - error
    - 4
  linebreak-style:
    - error
    - unix
  quotes:
    - off
    - double
  semi:
    - error
    - always
  no-unused-vars:
    - error
    - varsIgnorePattern: "^_$"
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
/*
  no-unused-vars:
    - error
    - varsIgnorePattern: "^_$"
*/
buttonActions($('#next-word'), _ => changeWord(1), _ => changeWord(10));
```

```bash
$ eslint *.js
```

**What did you expect to happen?**
I expected it to report that `argsIgnorePattern` is not specified, which caused this error.

**What actually happened? Please include the actual, raw output from ESLint.**
It told me about `varsIgnorePattern`, which confused me and made it difficult to understand the problem.
```
error  '_' is defined but never used. Allowed unused vars must match /^_$/
```

**Comment by platinumazure** [comment]
[Confirmed](https://eslint.org/demo/#eyJ0ZXh0IjoiLyogZXNsaW50IG5vLXVudXNlZC12YXJzOiBbXCJlcnJvclwiLCB7IFwidmFyc0lnbm9yZVBhdHRlcm5cIjogXCJeXyRcIiB9XSAqL1xyXG5idXR0b25BY3Rpb25zKCQoJyNuZXh0LXdvcmQnKSwgXyA9PiBjaGFuZ2VXb3JkKDEpLCBfID0+IGNoYW5nZVdvcmQoMTApKTsiLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOjgsInNvdXJjZVR5cGUiOiJzY3JpcHQiLCJlY21hRmVhdHVyZXMiOnt9fSwicnVsZXMiOnt9LCJlbnYiOnt9fX0=) in the demo.

**Comment by akouryy** [comment]
I'm going to work on this.
